x-pack/plugins/reporting/public/redirect/redirect_app.tsx

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
i don't think we need the redirect app. We would load this app only inside chromium (when generating the report), its not something we plan to share with users in any manner right ?

if that is the case, i think we should use redirect endpoint that url service provides. It was mentioned that its limited by 2MB limit and that this is not enough for dashboards. But we already have that limitation today and i think we should not try to solve it in this PR but rather by future improvements to the URL service.

Reply:
The primary reason for the redirect app is to get around URL length limitations. I think we can further simplify it - for reports we can inject the report state using chromium APIs - but I'm not sure how we could improve the URL service to bridge state transfer from server to client outside of URLs (open to input here).

The way I was thinking about it was that this redirect app is a wrapper around the functionality inside of the share redirect app. The change I proposed is to export the `navigate` step as an API from `share` and call that so that we can pass the value in via JavaScript instead of through the URL. This frees us up to get the state into JS however we might need to without introducing complexity specific to reporting into `share`.

Additionally, in future, in the reporting UI we should give users the ability to navigate to where a report is being generated. For that case we would also need something like a reporting redirect app (unless we can improve URL service in some way to cater for this). This would prevent users from needing to go to the Kibana server logs if they no longer have the URL at hand which I think is nice debugging improvement.

Let me know what you think regarding improvements to the URL service or if anything I said is unclear.

Reply:
with url service we plan to add compression support, and we might also add POST support to address the url length limitation.

what is missing from url service to allow redirecting to where report was generated from ? isn't that just allowing user to open that same link that chromium would open ? or maybe we need to store another locatorParams if we want to get to a different view.

Reply:
> with url service we plan to add compression support

This could work, depending on the compression ratio and testing with real-world data (speaking to support to try and get some data).

> we might also add POST support to address the url length limitation

I think this would be a comprehensive solution. A little bit of googling, I _think_ we can do this with puppeteer (open a page with a POST https://github.com/puppeteer/puppeteer/issues/669), but will have to test as it might be fickle. Also, I might be wrong here, but is it supported to access the data posted to the browser from client JS?

> what is missing from url service to allow redirecting to where report was generated from ?

the main challenge is that we need a way to pass full app state back to the client. this may not live in the URL in Kibana (and might be too big when serialised, compression will help but depends on level of reduction - we should test with real-world, big examples).

If possible, opening a URL by POSTing with some state would be the best solution IMO.

Otherwise I think a reporting-specific redirect app is the simplest solution and also offers the future possibility of letting the user's browser be redirected to page in Kibana with full app state.

Let me know what you think!